documentation/modules/exploit/multi/iiop/cve_2023_21839_weblogic_rce.md

```diff
@@ -0,0 +1,120 @@
+## Vulnerable Application
+
+### Description
+Oracle Weblogic 12.2.1.3.0, 12.2.1.4.0 and 14.1.1.0.0 prior to the Jan 2023 security update are vulnerable to an unauthenticated
+remote code execution vulnerability due to a post deserialization vulnerability. This occurs when an attacker serializes
+a `ForeignOpaqueReference` class object, deserializes it on the target, and then post deserialization, calls the
+object's `getReferent()` method, which will make use of the `ForeignOpaqueReference` class's `remoteJNDIName` variable,
+which is under the attackers control, to do a remote loading of the JNDI address specified by `remoteJNDIName` via
+the `lookup()` function.
```

Comment:
Might want to mention the current exploit only works on older JDKs, aka those that have not applied the mitigations mentioned at https://community.f5.com/t5/technical-articles/mitigating-new-gadget-leveraging-jndi-injection-into-remote-code/ta-p/282849 r.e the `trustURLCodebase` property.

According to that URL there is a bypass at https://gist.github.com/TheGrandPew/748ac740698511975eaeed5d77ecb2d9 which seems to bypass JNDI mitigations using `org.yaml.snakeyaml.Yaml`.

There are also more bypasses and more modern code at https://github.com/pimps/JNDI-Exploit-Kit which seems to suggest some more JNDI bypasses using Tomcat 8+ and Groovy, or Tomcat 8+ or SpringBoot 1.2.x+, so long as they are in the class path.

Might be an idea to update this to take those into account to affect more systems, but not necessarily a blocker right now.

Reply:
Update https://www.veracode.com/blog/research/exploiting-jndi-injections-java is also worth a read if we want to go ahead and implement this, breaks it down a lot more cleanly.